newm-server/src/main/kotlin/io/newm/server/features/marketplace/repo/MarketplaceRepositoryImpl.kt

```diff
@@ -68,6 +100,118 @@ internal class MarketplaceRepositoryImpl(
         }
     }
 
+    override suspend fun generateOrderAmount(request: OrderAmountRequest): OrderAmountResponse {
+        log.debug { "generateOrderAmount: $request" }
+
+        val sale = transaction { MarketplaceSaleEntity[request.saleId] }
+        require(request.bundleQuantity <= sale.availableBundleQuantity) { "Not enough bundles available for sale" }
+
+        val minIncentiveAmount = configRepository.getLong(CONFIG_KEY_MARKETPLACE_MIN_INCENTIVE_AMOUNT)
+        val incentiveAmount =
+            request.incentiveAmount?.also {
+                require(it < minIncentiveAmount) { "Not enough incentive" }
```

Comment:
Add `$it` value to the error message so we know exactly how the user screwed up.